code/handlers/trackservers.q

```diff
@@ -331,15 +331,22 @@ startup:{
 // Check if required processes all connected
 reqprocsnotconn:{[requiredprocs] 
     not all requiredprocs in exec proctype from .servers.SERVERS where .dotz.liveh[w]
-  }
+  };
 
 // Block process until all required processes are connected
-startupdependent:{[requiredprocs;timeintv]
-    while[.servers.reqprocsnotconn[requiredprocs];
-          .os.sleep[timeintv];
-          .servers.startup[]
-         ]
-  }
+startupdepcycles:{[requiredprocs;timeintv;cycles] 
+  n:0;                                                                                                                  //variable used to check how many cycles have passed
+  while[.servers.reqprocsnotconn requiredprocs;                                                                         //check if requiredprocs are running
+    n+:1;                                                                                                               //cycle counter
+    if[n>cycles;
+      b:((),requiredprocs)except(),exec proctype from .servers.SERVERS where .dotz.liveh w;
+      .lg.e[`connectionreport;raze string[.proc.procname]," cannot connect to ",
```

Comment:
This shouldn't be an error - should just be an .lg.o as otherwise processes that use this will fail to start. 

Also, I think you should put .servers.startup before this, otherwise every time it is executed you'll get an error message (even if it works correctly first time)